driver-core/src/main/java/com/datastax/driver/core/AbstractMonotonicTimestampGenerator.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Base implementation for monotonic timestamp generators.
+ * <p/>
+ * Implementors should guarantee that returned timestamps will always be monotonically
+ * increasing, even if multiple updates happen under the same millisecond.
+ * <p/>
+ * Note that to guarantee monotonicity, if more than 1000 timestamps are generated
+ * in the same millisecond, or in the event of a system clock skew, implementors are allowed to
+ * return timestamps that drift out in the future.
+ * <p/>
+ * The accuracy of the sub-millisecond part of the generated timestamps is largely dependent on the
+ * granularity of the operating system's clock.
+ * <p/>
+ * Generally speaking, this granularity is the millisecond, and
+ * the sub-millisecond part is simply a counter that gets incremented
+ * until the next clock tick, as provided by {@link System#currentTimeMillis()};
+ * on Linux systems however, it is possible to have a better granularity by using a JNA
+ * system call to {@code clock_gettime}. To benefit from this system call, set the system
+ * property {@code com.datastax.driver.USE_NATIVE_CLOCK} to {@code true}.
+ * <p/>
+ * This implementation is inspired by {@code org.apache.cassandra.service.ClientState#getTimestamp()}
+ */
+public abstract class AbstractMonotonicTimestampGenerator implements TimestampGenerator {
```

Comment:
Note that this class was previously package private; I'm proposing to make it public so that the documentation here is visible and valid for the two subclasses shipped with the driver: AtomicMonotonicTimestampGenerator and ThreadLocalMonotonicTimestampGenerator.